src/tools/compiletest/src/util.rs

```diff
@@ -211,6 +213,55 @@ pub fn logv(config: &Config, s: String) {
     }
 }
 
+/// Parses the output of `rustc --print cfg` into a [`HashMap`] containing
+/// the cfg values for each name. If the name is present with no value,
+/// it is treated as having a value of "".
+pub fn parse_rustc_cfg(rustc_output: String) -> HashMap<String, Vec<String>> {
+    let mut map = HashMap::new();
+
+    for line in rustc_output.lines() {
+        if let Some((name, value)) = line.split_once('=') {
+            let normalized_value = value.trim_matches('"');
+            cfg_add(&mut map, name, normalized_value);
+        } else {
+            cfg_add(&mut map, line, "");
+        }
+    }
+
+    return map;
+}
+
+/// Adds the given name and value to the provided cfg [`HashMap`]. If the `name` already
+/// points to a vector, this adds `value` to the vector. If `name` does not point
+/// to a vector, this adds a new vector containing only `value` to the [`HashMap`].
+fn cfg_add(map: &mut HashMap<String, Vec<String>>, name: &str, value: &str) {
+    let name = name.to_string();
+    let value = value.to_string();
+
+    if let Some(values) = map.get_mut(&name) {
+        values.push(value.to_string());
+    } else {
+        map.insert(name, vec![value.to_string()]);
+    }
```

Comment:
Fwiw I usually write this sort of thing with the entry API, entry(key).or_default().push(...).

Reply:
I can do that. It's a bit cleaner.